fix(ticket-deserialization): handle JSON deserialization errors for Ticket.class

tomrist9 opened the pull request:
- Added missing 	icketClass field in Ticket class to match JSON structure.
- Implemented no-argument constructor in Ticket class to enable Jackson deserialization.
- Configured Jackson to ignore unknown JSON fields using @JsonIgnoreProperties annotation.
- Updated loadTicketsFromResources method to support JSON arrays and single objects.

These changes ensure robust handling of JSON deserialization and prevent errors from unrecognized fields or missing constructors.

tomrist9 committed:
fix(ticket-deserialization): handle JSON deserialization errors for Ticket class

- Added missing 	icketClass field in Ticket class to match JSON structure.
- Implemented no-argument constructor in Ticket class to enable Jackson deserialization.
- Configured Jackson to ignore unknown JSON fields using @JsonIgnoreProperties annotation.
- Updated loadTicketsFromResources method to support JSON arrays and single objects.

These changes ensure robust handling of JSON deserialization and prevent errors from unrecognized fields or missing constructors.

Darya-Drutsko commented on andersen-homework/src/main/java/Main.java:
```diff
@@ -0,0 +1,23 @@
+
+        try {
+            // Load tickets from JSON file
+            List<Ticket> tickets = validator.loadTicketsFromFile("tickets.json");
```
save file name in separate constant(also could be stored in config file)

tomrist9 replied:
I have added ticket filename as a configurable property in application.properties

Darya-Drutsko replied:
Don't see the result in the pr

tomrist9 replied:
check it again teacher please

Darya-Drutsko commented on andersen-homework/src/main/java/org/model/Ticket.java:
```diff
@@ -0,0 +1,51 @@
+
+public class Ticket {
+
+    public String startDate;
```
why public modifier here?

tomrist9 replied:
when i selected LocalDate as a modifier , I get this error in validate method Cannot resolve method 'isEmpty' in 'LocalDate' therefore I used string as a modifier.

Darya-Drutsko replied:
I mean why have you chosen here 'public' access modifier, not 'private'?

tomrist9 replied:
i did not do it intentionally, so I ll change it to private

tomrist9 replied:
because when i chose private, i get this error in validate method 'ticketType' has private access in 'org.model.Ticket'

tomrist9 replied:
now i have solved it with get method

Darya-Drutsko replied:
Don't see the result in the pr

tomrist9 replied:
check it again please, teacher

Darya-Drutsko commented on andersen-homework/src/main/java/org/util/BusTicketValidator.java:
```diff
@@ -0,0 +1,93 @@
+
+    public boolean validateTickets(Ticket ticket) {
+        List<String> violations = new ArrayList<String>();
+        if (List.of("DAY", "WEEK", "MONTH").contains(ticket.ticketType) && (ticket.startDate == null || ticket.startDate.isEmpty())) {
```
Seems like DAY", "WEEK", "MONTH could be an enum :)

tomrist9 replied:
done

Darya-Drutsko commented on andersen-homework/src/main/java/org/util/BusTicketValidator.java:
```diff
@@ -0,0 +1,93 @@
+    private Map<String, Integer> violationCounts = new HashMap<>();
+
+    public BusTicketValidator() {
+        violationCounts.put("start date", 0);
```
Also I see that "start date" duplicated many times here, let's create constant for such values

Darya-Drutsko replied:
Add please constants for duplicated values such as "start date", "ticket type", "price"

tomrist9 replied:
have created constant

Darya-Drutsko reviewed (commented):
Good job!

tomrist9 committed:
fix: Adjust validateTickets method to support LocalDate for startDate

- Updated validations to use `null` checks for `startDate` instead of `isEmpty()`.
- Simplified logic by removing string-based parsing and directly using `LocalDate`.
- Ensured proper validation for ticket type, price, and start date in the future.

tomrist9 committed:
refactor: Change startDate from String to LocalDate in Ticket class

- Updated the startDate field to use LocalDate for better date handling.
- Replaced isEmpty() checks with
ull checks to accommodate the new data type.
- Removed unnecessary date parsing logic since startDate is now a LocalDate.

tomrist9 committed:
made changes in validate method

tomrist9 committed:
feat: Add ticket filename as a configurable property in application.properties

tomrist9 committed:
added get method

- Adjusted indentation and spacing for improved readability
- Organized imports and removed unused ones
- Added proper line breaks and consistent braces
- Ensured fields and methods follow naming conventions
- Improved documentation with Javadoc comments for clarity

tomrist9 committed:
fixid issues with filepath and applied formatting to all classes

tomrist9 committed:
Add .idea folder to .gitignore

tomrist9 committed:
fixid issues with filepath and applied formatting to all classes

Darya-Drutsko reviewed (approved):
Approved, but remove please .idea folder from git and return back files, related to previous homeworks :)
After this changes merge the pull request without additional review from my side

tomrist9 committed:
Merge branch 'main' into loop-task

# Conflicts:
#	.idea/compiler.xml
#	andersen-homework/src/main/java/org/example/Main.java

tomrist9 committed:
Ignore .idea folder
